frameworks/FSharp/frank/src/App/Program.fs

```diff
@@ -45,21 +45,21 @@ type MemoryStreamCache =
 let inline contentLength x = new Nullable<int64> ( int64 x )
 
 let json' data : HttpContext -> Task<unit> =
-    let bytes = Utf8Json.JsonSerializer.Serialize(data)
     fun ctx -> 
         ctx.Response.ContentLength <- contentLength bytes.Length
         ctx.Response.ContentType <- "application/json"
         ctx.Response.StatusCode <- 200
+        let bytes = Utf8Json.JsonSerializer.Serialize(data)
         task {
             do! ctx.Response.Body.WriteAsync(bytes, 0, bytes.Length)
         }
 
 let text' (msg:string): HttpContext -> Task<unit> =
-    let bytes = System.Text.Encoding.UTF8.GetBytes(msg)
     fun ctx ->
         ctx.Response.ContentLength <- contentLength bytes.Length
         ctx.Response.ContentType <- "text/plain"
         ctx.Response.StatusCode <- 200
+        let bytes = System.Text.Encoding.UTF8.GetBytes(msg)
```

Comment:
Just to be clear, the plaintext route does not have the same requirements.

Reply:
Ah, thanks!